### ISSUE TITLE
`fs.createWriteStream` overwriting behavior differs from node.js

### ISSUE THREAD

**Issue by noraj** [oven-sh/bun#13328]
### What version of Bun is running?

1.1.22

### What platform is your computer?

Linux 6.10.4-arch2-1 x86_64

### What steps can reproduce the bug?

Building whatever mozilla extension with https://github.com/mozilla/web-ext

- bun with nodejs: `bun run web-ext build --overwrite-dest -a ./dist -s ./my-extension`
- forcing native bun: `bun run --bun web-ext build --overwrite-dest -a ./dist -s ./my-extension`

Example repo:
- clone https://github.com/noraj/firefox-extension-arch-search
- `bun run buildAll`

### What is the expected behavior?

node.js 22.6.0 behaves like this

![image](https://github.com/user-attachments/assets/eece4eb5-9895-432e-8b71-b83c67e6a39d)

it overwrites the files

### What do you see instead?

bun behaves like this:

![image](https://github.com/user-attachments/assets/19f09d08-d910-4f42-b70a-d557b7ad933c)

fails because the file already exists, refuse to overwrite

### Additional information

web-ext code is like this:

![image](https://github.com/user-attachments/assets/a399e82e-ddfc-4127-bc05-5ca4fbb08d9d)

**Comment by RiskyMH** [comment]
Yeah your idea was correct, bun throws error on `createWriteStream` vs node if the `error` event is listened to uses it instead of error.

```js
import { createWriteStream } from 'fs';

// bun errors here as file exists
const stream = createWriteStream("./abcd.txt", { flags: 'wx' });

// node receives the error event instead
stream.on("error", () => console.log("error :("))
```

**Comment by RiskyMH** [comment]
I believe this is fixed in canary now (due to `node:fs` compat improvements). I'll close this for now as fixed for me, let me know if its still broken for you and I can reopen.
